public/app/features/dashboard/components/GenAI/GenAIButton.tsx

```diff
@@ -1,17 +1,19 @@
 import { css } from '@emotion/css';
-import React, { useEffect, useState } from 'react';
+import React from 'react';
 
 import { GrafanaTheme2 } from '@grafana/data';
+import { llms } from '@grafana/experimental';
 import { Button, Spinner, useStyles2, Link, Tooltip } from '@grafana/ui';
 
-import { Message, generateTextWithLLM, isLLMPluginEnabled } from './utils';
+import { useOpenaiStreamHook } from './hooks';
+import { OPEN_AI_MODEL } from './utils';
 
 export interface GenAIButtonProps {
   text?: string;
   loadingText?: string;
   onClick?: (e: React.MouseEvent<HTMLButtonElement>) => void;
-  messages: Message[];
-  onReply: (response: string, isDone: boolean) => void;
+  messages: llms.openai.Message[];
```

Comment:
Would you re-export the Message type from utils to not import experimental everywhere? 🤔

Reply:
Done

Reply:
This caused a bunch of merge conflicts with main for some reason, though